src/AWS.Deploy.CLI/ServerMode/Controllers/RecipeController.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using AWS.Deploy.CLI.ServerMode.Models;
+using AWS.Deploy.Orchestration;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace AWS.Deploy.CLI.ServerMode.Controllers
+{
+    [Produces("application/json")]
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class RecipeController : ControllerBase
+    {
+        /// <summary>
+        /// Gets a summary of the specified Recipe.
+        /// </summary>
+        [HttpGet("{recipeId}")]
+        [SwaggerOperation(OperationId = "GetRecipe")]
+        [SwaggerResponse(200, type: typeof(RecipeSummary))]
+        public IActionResult GetRecipe(string recipeId)
```

Comment:
I think we should take the project path as an optional parameter coming in as a header. That way if the project path is available we can search for the recipes based on the context of a specific project. That way when we save CDK projects we can lookup recipes the upcoming json mapping file.  We don't need to do anything with the value till we have the save CDK project feature implemented but I think we should get the API define fully from the start.

Reply:
I have added the optional query parameter.